### PR TITLE
♻️ Rename amp-document-recommendations to amp-next-page

### PR DESCRIPTION
**Pull request by peterjosling**
As agreed in the naming spreadsheet.

🎉🎉🎉

### CODE REVIEW

File: extensions/amp-next-page/validator-amp-next-page.protoascii [hunk 1]
```diff
@@ -13,43 +13,43 @@
...
   html_format: AMP
   tag_name: "SCRIPT"
-  spec_name: "amp-document-recommendations extension .json configuration"
-  requires_extension: "amp-document-recommendations"
-  mandatory_parent: "AMP-DOCUMENT-RECOMMENDATIONS"
+  spec_name: "amp-next-page extension .json configuration"
+  requires_extension: "amp-next-page"
+  mandatory_parent: "AMP-NEXT-PAGE"
   attrs: {
     name: "recommendations"
```
Comment (honeybadgerdontcare) [inline extensions/amp-next-page/validator-amp-next-page.protoascii]: should this become `pages` instead of `recommendations`?

File: gulpfile.js [hunk 2]
```diff
@@ -102,7 +102,7 @@ declareExtension('amp-compare-slider', '0.1');
 declareExtension('amp-consent', '0.1', {hasCss: true});
 declareExtension('amp-crypto-polyfill', '0.1');
 declareExtension('amp-dailymotion', '0.1');
-declareExtension('amp-document-recommendations', '0.1', {hasCss: true});
+declareExtension('amp-next-page', '0.1', {hasCss: true});
```
Comment (honeybadgerdontcare) [inline gulpfile.js]: alphabetize

File: extensions/amp-next-page/amp-next-page.md [hunk 3]
```diff
@@ -51,19 +51,19 @@ limitations under the License.
 
 ## Behavior
 
-Given a list of recommended documents, `amp-document-recommendations` tries to load them after the current document, provding an infinite-scroll type experience.
+Given a list of recommended documents, `amp-next-page` tries to load them after the current document, providing an infinite-scroll type experience.
```
Comment (honeybadgerdontcare) [inline extensions/amp-next-page/amp-next-page.md]: should `recommended documents` become just `pages`?

**Comment by cvializ** [comment]
I'll defer to @honeybadgerdontcare's review. Do you have a link to the spreadsheet? I looked but couldn't find it

File: extensions/amp-next-page/validator-amp-next-page.protoascii (hunk 1 above)
Comment (peterjosling) [reply extensions/amp-next-page/validator-amp-next-page.protoascii]: Yup, makes more sense. Done.

File: gulpfile.js (hunk 2 above)
Comment (peterjosling) [reply gulpfile.js]: Done.

File: extensions/amp-next-page/amp-next-page.md (hunk 3 above)
Comment (peterjosling) [reply extensions/amp-next-page/amp-next-page.md]: Done.

**Comment by peterjosling** [comment]
Ready to merge.

**Pull request merged**